test/extensions/filters/http/jwt_authn/filter_config_test.cc

```diff
@@ -159,6 +160,70 @@ TEST(HttpJwtAuthnFilterConfigTest, FindByFilterState) {
               nullptr);
 }
 
+TEST(HttpJwtAuthnFilterConfigTest, FindByRequiremenMap) {
+  const char config[] = R"(
+providers:
+  provider1:
+    issuer: issuer1
+    local_jwks:
+      inline_string: jwks
+  provider2:
+    issuer: issuer2
+    local_jwks:
+      inline_string: jwks
+requirement_map:
+  r1:
+    provider_name: provider1
+  r2:
+    provider_name: provider2
+)";
+
+  JwtAuthentication proto_config;
+  TestUtility::loadFromYaml(config, proto_config);
+
+  NiceMock<Server::Configuration::MockFactoryContext> context;
+  auto filter_conf = FilterConfigImpl::create(proto_config, "", context);
+
+  PerRouteConfig per_route;
+  const Verifier* verifier;
+  std::string error_msg;
+
+  per_route.Clear();
+  per_route.set_bypass(true);
+  std::tie(verifier, error_msg) =
+      filter_conf->findPerRouteVerifier(PerRouteFilterConfig(per_route));
+  EXPECT_EQ(verifier, nullptr);
+  EXPECT_EQ(error_msg, EMPTY_STRING);
+
+  per_route.Clear();
+  per_route.set_requirement_name("r1");
+  std::tie(verifier, error_msg) =
+      filter_conf->findPerRouteVerifier(PerRouteFilterConfig(per_route));
+  EXPECT_NE(verifier, nullptr);
+  EXPECT_EQ(error_msg, EMPTY_STRING);
+
+  per_route.Clear();
+  per_route.set_requirement_name("r2");
+  std::tie(verifier, error_msg) =
+      filter_conf->findPerRouteVerifier(PerRouteFilterConfig(per_route));
+  EXPECT_NE(verifier, nullptr);
+  EXPECT_EQ(error_msg, EMPTY_STRING);
+
+  per_route.Clear();
+  per_route.set_requirement_name("wrong-name");
+  std::tie(verifier, error_msg) =
+      filter_conf->findPerRouteVerifier(PerRouteFilterConfig(per_route));
+  EXPECT_EQ(verifier, nullptr);
+  EXPECT_EQ(error_msg, "Wrong requirement_name: wrong-name. It should be one of [r1,r2]");
+
+  // Empty requirement_name
+  per_route.Clear();
+  std::tie(verifier, error_msg) =
+      filter_conf->findPerRouteVerifier(PerRouteFilterConfig(per_route));
+  EXPECT_EQ(verifier, nullptr);
+  EXPECT_EQ(error_msg, "Wrong requirement_name: . It should be one of [r1,r2]");
```

Comment:
is it better to annotate the requirement_name to require it is non-empty?

Reply:
This will not happen.  requirement_name has validation rule requiring non-empty.  This unit-test skip the proto validation.